packages/web-components/src/components/va-button/va-button.tsx

```diff
@@ -76,17 +76,20 @@ export class VaButton {
   })
   componentLibraryAnalytics: EventEmitter;
 
-  private handleClick = (e: MouseEvent): void => {
-    // TODO: remove console.log
-    console.log(e);
-    if (this.disableAnalytics) return;
-    this.componentLibraryAnalytics.emit({
-      componentName: 'va-button',
-      action: 'click',
-      details: {
-        // TODO: add analytics event details
-      },
-    });
+  private handleClick = (): void => {
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-button',
+        action: 'click',
+        details: {
+          event: 'cta-button-click',
+          'button-type': this.secondary ? 'secondary' : 'primary',
+          'button-click-label': this.getButtonText(),
+          'button-background-color': this.secondary ? 'white' : '#0071bb',
```

Comment:
I'm not sure why we want to send the background color, but hardcoding a hex here seems like something that will bite us in the future if the color changes.

Reply:
@bkjohnson This was added to the PR before I took over the analytics ticket so I don't know the reason or the history behind the default color but following the tickets related to this, I can see that background color is part of the data layer specs here:

https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/design-system-components